app/controllers/organizations/invitations_controller.rb

```diff
@@ -1,20 +1,25 @@
 class Organizations::InvitationsController < Devise::InvitationsController
   include OrganizationScopable
 
-  before_action :require_organization_admin, only: [:new, :create]
   layout "dashboard", only: [:new, :create]
 
   def new
+    authorize! StaffAccount, context: {organization: Current.organization},
+      with: Organizations::InvitationPolicy
+
```

Comment:
Little confused about this part - so in the policy it's checking if this current_user has `:invite_staff` permission. How does it know which set of roles to check for? How is `StaffAccount` being used here?

Reply:
I found the controllers that do not match with a model the hardest to write policies for, like this one. 

You are correct. There is no existing record to check, so all this authorization is doing is checking that the user has the `:invite_staff` permission, and that the user's organization matches with the current organization (via the `pre_check :verify_organization!`.

`StaffAccount` does not actually get used here. We could have passed anything, a string `"invitation"` or symbol `:invitation`. However, I felt it was best to pass the class of the model that is associated with the action. So in this case, we are building a new `StaffAccount` so I passed that class to the policy. 

Action Policy by default always passes a user context from `current_user`. The `User` model handles `permission?` via the `authorizable` concern. In this way, policies do not know which roles to check for. All they know is that user's reply to `permission?` 

I personally really like this pattern and don't think policies need to know about roles at all. All roles do for this implementation, is store a list of permissions. Then, the policies check for those permissions. In the future if it was desired, we could even add the permissions as an attribute to users so that user's could have individual sets of permissions that could be unique. Note, I wrote `authorizable` so the permissions and roles implementation is not part of Action Policy.

Hope that helps. Let me know if I can further clarify

Reply:
Thanks for the explanation!